subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -969,6 +1197,21 @@ int nrf_cloud_modem_info_json_encode(const struct nrf_cloud_modem_info *const mo
 	cJSON_Delete(tmp);
 	return err;
 }
+#else
+int nrf_cloud_modem_info_json_encode(const struct nrf_cloud_modem_info *const mod_inf,
+				     cJSON *const mod_inf_obj)
+{
+	ARG_UNUSED(mod_inf_obj);
+
+	if ((mod_inf->device == NRF_CLOUD_INFO_SET ||
+	     mod_inf->sim == NRF_CLOUD_INFO_SET ||
+	     mod_inf->network == NRF_CLOUD_INFO_SET)) {
+		LOG_ERR("CONFIG_MODEM_INFO is not enabled, unable to set device info");
+		return -EACCES;
+	}
+	return 0;
```

Comment:
i believe the calls to `encode_info_item_cs` should still be here in case this function is used to clear an item with `NRF_CLOUD_INFO_CLEAR`?

Reply:
done